packages/marshaller/src/ddl2/activities/form.ts

```diff
@@ -139,14 +143,14 @@ export class Form extends Datasource {
         this.validFields().map(f => {
             switch (f.type()) {
                 case "boolean":
-                    retVal[f.fieldID()] = f.value_exists() ? f.value() : f.default();
+                    retVal[f.fieldID()] = f.calcValue();
                     break;
                 case "number":
-                    retVal[f.fieldID()] = f.value_exists() ? f.value() : f.default();
+                    retVal[f.fieldID()] = f.calcValue();
                     break;
                 case "string":
                 default:
-                    retVal[f.fieldID()] = f.value_exists() ? f.value() : f.default();
+                    retVal[f.fieldID()] = f.calcValue();
                     break;
             }
```

Comment:
This could be:
```js
switch(f.type()){
    case "boolean":
    case "number":
    case "string":
    default:
        retVal[f.fieldID()] = f.calcValue();
}
```